CodeU_Assignement_3/src/codeu_assignement_3/SimpleDictionary.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package codeu_assignement_3;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import static javafx.scene.input.KeyCode.T;
+
+/**
+ *
+ * @author Karine
+ * Simple dictionary only for testing
+ * 
+ * Consider using mocked dictionary instead of this implementation
+ * 
+ */
+public class SimpleDictionary implements KemDictionary 
+{
+    /* Dictionary as an array */
+    private final ArrayList<String> m_dict = new ArrayList<>(); 
+    
+    
+    /* Add all in BFS order (i.e., Breadth-First Traversal of a Tree) */

```

Comment:
It is possible to see the search as a tree ("c" is a father of "ca"). Now it is the algorithm choice how to do it. Search for "c" with "a" and "u" and so on, or search "c", "ca", and "cat" and then backtracking to search "cu" and "cut". If you backtrack then it is DFS style, if you do it by length (all that starts with "c" of length 2, then all that stats with "ca" of length of 3 and so on, then it is BFS style. But I can remove this comment. np.